quickwit/quickwit-search/src/scroll_context.rs

```diff
@@ -0,0 +1,234 @@
+// Copyright (C) 2023 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::ops::Range;
+use std::str::FromStr;
+use std::sync::Arc;
+use std::time::Duration;
+
+use anyhow::Context;
+use base64::prelude::BASE64_STANDARD;
+use base64::Engine;
+use quickwit_common::uri::Uri;
+use quickwit_metastore::SplitMetadata;
+use quickwit_proto::{LeafSearchResponse, PartialHit, SearchRequest};
+use serde::{Deserialize, Serialize};
+use tokio::sync::RwLock;
+use ttl_cache::TtlCache;
+use ulid::Ulid;
+
+use crate::service::SearcherContext;
+use crate::ClusterClient;
+
+/// Maximum capacity of the search after cache.
+///
+/// For the moment this value is hardcoded.
+/// TODO make configurable.
```

Comment:
can you open an issue on this?

Also, I don't understand the comment "
/// Assuming a search context of 1MB, this can
/// amount to up to 1GB.
"

Are you talking about the `scroll context`? If not, what is the `search context`?
And another question where the 1000 factor comes from (1MB -> 1GB)?